### ISSUE TITLE
How to use the functions in torch._C._THCUNN? 

### ISSUE THREAD

**Issue by mingminzhen** [pytorch/pytorch#11974]
Hi,
i want to use the functions in torch._C._thcunn. Is there any docs for that?
what’s the meaning of the parameters?
i find the C code for CudaSpatialFullDilatedConvolution_updateGradInput
```
void THNN_(SpatialFullDilatedConvolution_updateOutput)(
           THCState *state,
           THCTensor *input,
           THCTensor *output,
           THCTensor *weight,
           THCTensor *bias,
           THCTensor *columns,
           THCTensor *ones,
           int kW, int kH,
           int dW, int dH,
           int padW, int padH,
           int dilationW, int dilationH,
           int adjW, int adjH)
```

but i don't know the meaning of the ```state```? How can i specify that variable in python?

**Comment by SsnL** [comment]
They aren't public API and maybe removed in future. Why not just using the public API?

**Comment by zou3519** [comment]
As @SsnL mentioned this API is not public. What layers exactly are you trying to implement?

**Comment by mingminzhen** [comment]
@zou3519 I want to specify the output tensor for convolution and batchnorm layer.

**Comment by ezyang** [comment]
For manual forwards-backwards, there now is torch.nn.grad

For specifying the output tensor, this is still a valid feature request, we still don't support this.
